### PR TITLE
[WIP] Drag companion window to bottom (and stacking)

### PR DESCRIPTION
**Pull request by charbugs**
#1760 
#1754

### CODE REVIEW

File: src/containers/Window.js [hunk 1]
```diff
@@ -12,8 +13,41 @@ const mapStateToProps = ({ manifests, windows }, props) => ({
...
+  companionAreaBottom: {
+    flex: '0',
+  },
+  thumbnailArea: {
+    flex: '0',
+  },
+};
+
 const enhance = compose(
   connect(mapStateToProps),
```
Comment (christopher-johnson) [inline src/containers/Window.js]: withStyles needs to be before connect here

File: src/components/CompanionWindow.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import Divider from '@material-ui/core/Divider';
 import CloseIcon from '@material-ui/icons/Close';
+import OpenWith from '@material-ui/icons/OpenWith';
```
Comment (jkeck) [inline src/components/CompanionWindow.js]: Can we make this `OpenWithSharp` to match other Sharp icons now that we've standardized?

File: src/state/actions/window.js [hunk 3]
```diff
@@ -117,6 +116,18 @@ export function closeCompanionWindow(windowId, companionWindowId) {
...
 
+/**
+* Move companion window between right and bottom area
+* @param {String} companionWindowId
+*/
+export function toggleAreaOfCompanionWindow(companionWindowId) {
+  return (dispatch, getState) => {
+    const position = getState()
+      .companionWindows[companionWindowId].position === 'right' ? 'bottom' : 'right';
+    dispatch(updateCompanionWindow(companionWindowId, { position }));
```
Comment (jkeck) [inline src/state/actions/window.js]: It looks like we may be missing a unit test for this function, would it be possible/easy to add one?

File: src/components/CompanionWindow.js (hunk 2 above)
Comment (jvine) [reply src/components/CompanionWindow.js]: Or can we use the drag icon listed in #1947 ?   DragIndicatorSharp

**Comment by jvine** [comment]
The issue with multiple windows covering the sidebar might be partly related to #1934. The sidebar should be on the same level as the canvas and companion windows, and technically they should be able to shove each other over/make each other relatively smaller as needed - but there's still a boundary where there are too many windows to be functional. And it's complicated by the fact that the windows will be able to expand the window size in Freeform mode. 

Possibly approach is to limit the maximum area that can be used by companion windows (or maintain a minimum area of visible canvas), and once that limit is reached, block any further companion windows from being opened. Should this be a separate ticket?

File: src/components/CompanionWindow.js (hunk 2 above)
Comment (jkeck) [reply src/components/CompanionWindow.js]: Sure! I suppose I misunderstood 
> I don't think this icon is directly needed yet.

in https://github.com/ProjectMirador/mirador/pull/1967#issuecomment-467540566

**Comment by jkeck** [comment]
I think it's fine to push that to a separate issue (assuming we're okay w/ the interim state where that menu can become inaccessible)

**Comment by aeschylus** [comment]
Compare to https://github.com/ProjectMirador/mirador/issues/1760

**Comment by charbugs** [comment]
@aeschylus 
What are the implications of this comment? :confused:

**Comment by jvine** [comment]
I can't see the preview on this one - I just get the blank white page. The screenshot @jkeck posted looks accurate in terms of layout but I'd like to verify.

**Comment by charbugs** [comment]
Hmm, strange. Work for me here. Is there an error message on the console?

**Comment by jvine** [comment]
Okay, after a few tries the preview randomly works, yay!

If this PR is just about the placement of multiple companion windows - stacking vertically, horizontally, and in relationship to the thumbnail nav - then the layout is accurate. Some later discussion (in a ticket or Slack, I don't recall) added to/superseded #1760:  the thumbnail nav is always at the bottom of the stack (or far right for vertical) and the full width (or height) of the entire window. 👍 

However:
1. what's implemented here is not a drag interaction but a side/bottom toggle. That means some other intended interactions need further specification, like reordering companion windows within a stack. 
2. there needs to be some HTML restructuring (which will apply to the sidebar as well) to make the header (title plus controls) a unit that is separate from the content. Will ticket this separately.
3. companion windows desperately need some kind of border to distinguish them from each other. They're incomprehensible without this. Curse MUI and its borderlessness!

What's here is accurate. But combining the visual design & HTML issues with the hidden sidebar issue above, and given @snydman's comments today - I think it's premature to merge this to master. @ggeisler do you agree?

**Comment by ggeisler** [comment]
Yeah, it seems like there is still quite a bit of fine-tuning needed, so rather than merging now, it might make more sense to first try to work in some of the details @jvine outlined above.

**Comment by charbugs** [comment]
Fine with me. So I will lable the PR as WIP.

**Comment by cbeer** [comment]
I think we've extracted all the behavior into separate PRs and this can be safely closed.

**Pull request closed without merging**